File: backend-java/src/main/java/com/carrental/model/Maintenance.java
package com.carrental.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@Document(collection = "maintenances")
public class Maintenance {

    @Id
    private String id;
    
    private String carId;
    
    private String ownerId;
    
    private String type; // Service, Repair, Inspection, Cleaning, Tire Change, etc.
    
    private String description;
    
    private Double cost;
    
    private String serviceProvider;
    
    private Date date;
    
    private Date nextServiceDate;
    
    private Integer mileage;
    
    private List<String> receipts;
    
    private String notes;
    
    private Date createdAt = new Date();
}

File: backend/models/Maintenance.js
const mongoose = require('mongoose');

const maintenanceSchema = new mongoose.Schema({
  car: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Car',
    required: true
  },
  owner: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  type: {
    type: String,
    required: true,
    enum: ['Service', 'Repair', 'Inspection', 'Cleaning', 'Tire Change', 'Other']
  },
  description: {
    type: String,
    required: true
  },
  cost: {
    type: Number,
    required: true,
    min: 0
  },
  serviceProvider: String,
  date: {
    type: Date,
    required: true
  },
  nextServiceDate: Date,
  mileage: Number,
  receipts: [{
    type: String
  }],
  notes: String,
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('Maintenance', maintenanceSchema);
